Piece of code output c compiler error

Function does not compile and outputs c compiler error instead of nim compiler error
### Example
https://play.nim-lang.org/#ix=2R5P

```nim
const DEBUG = true
  
type Matrix = seq[seq[int]]
  
proc needlemanWunsch(sequence1: string, sequence2: string, gap_penal: int8, match: int8, indel_penal: int8): bool =
  let seq1_len = sequence1.len
  let seq2_len = sequence2.len
  
  var grid: Matrix
  for i in sequence1:
    grid.add(newSeqOfCap[seq[int]](seq2_len))
  when DEBUG:
    echo "Length of first sequence ", seq1_len, "Length of first array index ", grid.len
  result = true
  

let a = needlemanWunsch("ABC", "DEFG", 1, 2, 3) 
```

### Current Output
Compiler error from the c compiler
```
CC: stdlib_assertions.nim
CC: stdlib_io.nim
CC: stdlib_system.nim
CC: needlemanWunsch.nim
/home/td/.cache/nim/needlemanWunsch_d/@mneedlemanWunsch.nim.c: In function 'needlemanWunsch__z9bNcYob7uj6yPdJ9cpn4lIQ':
/home/td/.cache/nim/needlemanWunsch_d/@mneedlemanWunsch.nim.c:206:150: error: request for member 'data' in something not a structure or union
 Nlvg((&grid), (tySequence__PSP8snSsRoFs9cDiya9bd7UQ*)nimNewSeqOfCap((&NTI__PSP8snSsRoFs9cDiya9bd7UQ_), ((NI) (seq2_len)))->data, ((tySequence__PSP8snSsRoFs9cDiya9bd7UQ*)nimNewSeqOfCap((&NTI__PSP8snSsRoFs9cDiya9bd7UQ_), ((NI) (seq2_len))) ? (tySequence__PSP8snSsRoFs9cDiya9bd7UQ*)nimNewSeqOfCap((&NTI__PSP8snSsRoFs9cDiya9bd7UQ_), ((NI) (seq2_len)))->Sup.len : 0));
                                                                                                                          ^~

/home/td/.cache/nim/needlemanWunsch_d/@mneedlemanWunsch.nim.c:206:376: error: request for member 'Sup' in something not a structure or union
 seq2_len))) ? (tySequence__PSP8snSsRoFs9cDiya9bd7UQ*)nimNewSeqOfCap((&NTI__PSP8snSsRoFs9cDiya9bd7UQ_), ((NI) (seq2_len)))->Sup.len : 0));
                                                                                                                          ^~

Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3   -I/home/td/nim/nim-1.4.4/lib -I/home/td/dev/nim/bionim/src -o /home/td/.cache/nim/needlemanWunsch_d/@mneedlemanWunsch.nim.c.o /home/td/.cache/nim/needlemanWunsch_d/@mneedlemanWunsch.nim.c' failed with exit code: 1
```

### Expected Output
A useful error message from the nim compiler

### Additional Information
Nim 1.4.4, linux
$ nim -v
Nim Compiler Version 1.4.4 [Linux: amd64]
Compiled at 2021-02-23
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 2ff517462bf8609b30e6134c96658aa7912b628a
active boot switches: -d:release

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -1468,6 +1468,7 @@ proc genNewSeqOfCap(p: BProc; e: PNode; d: var TLoc) =
       getSeqPayloadType(p.module, seqtype),
     ])
   else:
+    if d.k == locNone: d = getTemp(p, e.typ, needsInit=false) # bug #22560
     putIntoDest(p, d, e, ropecg(p.module,
                 "($1)#nimNewSeqOfCap($2, $3)", [
                 getTypeDesc(p.module, seqtype),

--- a/tests/collections/tseq.nim
+++ b/tests/collections/tseq.nim
@@ -223,3 +223,20 @@ for i in 0..100:
   var test = newSeqOfCap[uint32](1)
   test.setLen(1)
   doAssert test[0] == 0, $(test[0], i)
+
+
+# bug #22560
+doAssert len(newSeqOfCap[int](42)) == 0
+
+block: # bug #17197
+  type Matrix = seq[seq[int]]
+
+  proc needlemanWunsch(sequence1: string, sequence2: string, gap_penal: int8, match: int8, indel_penal: int8): bool =
+    let seq2_len = sequence2.len
+
+    var grid: Matrix
+    for i in sequence1:
+      grid.add(newSeqOfCap[seq[int]](seq2_len))
+    result = true
+
+  doAssert needlemanWunsch("ABC", "DEFG", 1, 2, 3)

